### DISCUSSION TITLE
Any way to bind specific network interface?

### DISCUSSION THREAD

**Discussion by june9713** [aio-libs/aiohttp#5504] [Q&A] [answered]
Hi.
we can use requests_toolbelt to bind specific network interface with requests moudule . 
Please read example code below.

```
from requests_toolbelt.adapters import source #need to install requests_toolbelt pip install requests_toolbelt
import requests

ss = requests.session()
iface = source.SourceAddressAdapter("##your netowrk address want to bind##")
ss.mount('http://', iface)
ss.mount('https://', iface))
ss.get("https://google.com") # it will request via network interface you choose.
```

Does aiohttp has Anyway to bind specific network interface to bind specific network interface?
Thanks.

**Comment by webknjaz** [comment, marked as answer]
I believe you should be able to specify a custom [`TCPConnector`] instance with a [`local_addr`] set. For example:
```python
import aiohttp


local_address = `localhost`, 12345
tcp_connector = aiohttp.TCPConnector(local_addr=local_address)

async with aiohttp.ClientSession(connector=tcp_connector) as http_session:
    async with http_session.get("https://google.com") as http_response:
        print(await http_response.text())
```

[`local_addr`]: https://docs.aiohttp.org/en/v3.7.4/client_reference.html#tcpconnector
[`TCPConnector`]: https://docs.aiohttp.org/en/v3.7.4/client_advanced.html#connectors

**Reply by june9713** [reply]
Oh really great!! Thanks!! You saved my many days.
local_address = "localhost", None is working. 
Thanks.

**Reply by Yuribtr** [reply]
This is really great, but sometimes (as in my case) there might be multiple interfaces with same IP. 
For example if we simultaneously connects to multiple WiFi AP's with same default settings or if we connect to multiple GoPro cameras (which give same default IP 10.5.5.100 to client). So this way is not acceptable for me (((
It will be very handy to use interface name (or number) such as "wlan0" or similar.

**Reply by june9713** [reply]
You can make Ip as static address :)

**Reply by Yuribtr** [reply]
Of course it is possible, but I don't want to have possible conflicts with other IP's in network.
